### DISCUSSION TITLE
Timestamp based versions in Maven Central

### DISCUSSION THREAD

**Discussion by aemaem** [graphql-java/graphql-java#2263] [Q&A]
What is the purpose of the timestamp based versions in maven central?
It is quite confusing that there are lots of timestamp versions and at the very bottom there are the few actually released versions: https://mvnrepository.com/artifact/com.graphql-java/graphql-java

**Comment by bbakerman** [comment]
Every merge to master creates a timestamped snapshot version.

So they are basically the latest code on the master branch at the time of merge.

Some people use this for early access to the latest code, including my team at Atlassian.

**Reply by aemaem** [reply]
Imo a version scheme like `0.0.0-20210325T000000.revnumber` or even the semver style (https://semver.org/#spec-item-10) would be less confusing and would fit into the correct sorting of versions. Anyway, thanks for clarification!
